app/search.py

```diff
@@ -494,6 +497,78 @@ def _query_fields(self, fields):  # pylint: disable=too-many-locals
                 f'{fields} "{preNonDigitAndPreDigit}"~5',
                 f'{fields} "{infNonDigitAndPreDigit}"'
             ]
+
+        # to improve the fuzzyness of the search result the search text has to be changed
+        # digit/text combinations are replaced with (digit|digit/text) p.e. 4a -> (4|4a)
+        # quorum operatar is used for query text fuzzy matching
+        if fuzzySearch:
+            # minimal length of single words that will be trimmed if only one word is in the query
+            # string
+            minLength = 5
+
+            def convert_if_digit(text):
+                """
+                replaces a keyword that begins with a digit the following sphinx query:
+                (digit|digit+rest)
+                examples:
+                    4a          -> (4|4a)
+                    342         -> (342)
+                    sometext    -> sometext
+                Args:
+                    text
+
+                Returns:
+                    sphinx query for digit and digit+rest if the keywords starts with a digit
+                    otherwise the input text will be returned unchanged
+                """
+                if isdigit(text):
+                    digit = re.findall(r'^\d+', text)[0]
+                    if digit != text:
+                        return f'({str(digit)}|{text})'
+                return text
+
+            def generate_prefixes(input_list, min_length=5):
+
+                def max_text_length(input_list):
```

Comment:
probably this method could be prefixed with `_`?

Reply:
done